src/coreclr/vm/precode_portable.hpp

```diff
@@ -31,6 +32,17 @@ class PortableEntryPoint final
 
 public:
     void Init(MethodDesc* pMD);
+
+    // Query methods for entry point state.
+    bool IsInitialized() const { return _pMD != nullptr; }
+    bool IsByteCodeCompiled() const { return IsInitialized() && _pInterpreterData != nullptr; }
```

Comment:
```suggestion
    bool IsByteCodeCompiled() const
    {
        _ASSERTE(IsInitialized());
        return _pInterpreterData != nullptr;
    }
```
We should never see uninitialized entrypoints in circulation. These should be asserts. (We can also assert that the canary is good in IsIniitalized. Maybe a better name for the method would be `IsValid`.